src/components/TodoList.jsx

```diff
@@ -1,64 +1,158 @@
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this)
+        this.sortTasks = this.sortTasks.bind(this)
+        this.setTaskInEditText = this.setTaskInEditText.bind(this)
+        this.setNewTaskText = this.setNewTaskText.bind(this)
     }
-
-    removeTask(text) {
+    // Set text of task in edit mode 
+    setTaskInEditText(taskInEditText) {
+        this.setState({
+            ...this.state,
+            taskInEditText: taskInEditText
+        })
+    }
+    // Change name of task
+    setNewTaskText(text, newText) {
+        const { tasks } = this.state
+        const id = this.getTaskNames().indexOf(text)
+        tasks[id].taskName = newText
+        this.setState({
+            ...this.state,
+            tasks: tasks
+        })
+    }
+    // Update state with sorted tasks
+    sortTasks(event) {
         const { tasks } = this.state;
+        const option = event.target.value
+        this.setState({
+            ...this.state,
+            tasks: tasks.sort(filterTasks(option)),
+            sortingOpt: option
+        });
+    }
 
+    // Array of names of tasks
+    getTaskNames() {
+        const taskNames = [];
+        for (const task of this.state.tasks) {
+            taskNames.push(task.taskName)
+        }
+        return taskNames
+    }
 
+    removeTask(todoPointName) {
+        const { tasks } = this.state;
+        tasks.splice(this.getTaskNames().indexOf(todoPointName), 1)
         const newState = {
-            ...this.state,
-            tasks: tasks.filter(elem => elem != text)
-        }
+            ...this.state
+        };
         this.setState(newState);
     }
 
     addTask() {
-        const { tasks } = this.state;
-
-        this.setState({
+        const { tasks, sortingOpt } = this.state;
+        const newTask = {
+            taskName: this.state.input,
+            date: new Date()
+        };
+        // Sort task as it added
+        const newTasks = tasks.concat(newTask).sort(filterTasks(sortingOpt));
+        const newState = {
             input: "",
-            tasks: tasks.concat(this.state.input)
-        });
+            tasks: newTasks
+        }
+        this.setState(newState);
     }
 
     onInputChange(event) {
         const text = event.target.value;
-        this.setState({ ...this.state, input: text });
+        this.setState({ ...this.state, input: text })
+    }
+
+    removeAllTasks() {
+        this.setState(
+            {
+                ...this.state,
+                tasks: []
+            }
+        )
+    }
+
+    mapTasks(mapArgument) {
+        return this.state.tasks.map(mapArgument)
+    }
+    // Key of task html element 
+    getKeyForTodoElem(todoPoint) {
+        return this.state.tasks.indexOf(todoPoint)
     }
 
     render() {
+        const sortingOpts = ["Sort by", "Newer", "Older", "Name asc", "Name desc"]
```

Comment:
This gets re-initialized every single time this component re-renders. Why not extract this to `utils.js`?